dotCMS/src/main/java/com/dotcms/storage/FileMetadataAPIImpl.java

```diff
@@ -272,45 +303,128 @@ public ContentletMetadata generateContentletMetadata(final Contentlet contentlet
+                return new Metadata(fieldVariableName, metadataMap);
+            }
+
+            if (forceGenerate) {
+                try {
+                    return generateContentletMetadata(contentlet).getFullMetadataMap()
+                            .get(fieldVariableName);
+                } catch (IOException e) {
+                    throw new DotDataException(e);
+                }
+            }
+        }
+        return null;
+
+    }
+
+
     /**
      * {@inheritDoc}
      * @param contentlet          {@link Contentlet}
      * @param fieldVariableName  {@link String}
      * @return
      */
     @Override
-    public Map<String, Serializable> getMetadataNoCache(final Contentlet contentlet,
+    public Metadata getFullMetadataNoCache(final Contentlet contentlet,
             final String fieldVariableName) throws DotDataException {
 
+        return getFullMetadataNoCache(contentlet, fieldVariableName, false);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param contentlet          {@link Contentlet}
+     * @param fieldVariableName  {@link String}
+     * @return
+     */
+    @Override
+    public Metadata getFullMetadataNoCacheForceGenerate(final Contentlet contentlet,
+            final String fieldVariableName) throws DotDataException{
+        return getFullMetadataNoCache(contentlet, fieldVariableName, true);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param contentlet          {@link Contentlet}
+     * @param fieldVariableName  {@link String}
+     * @param forceGenerate  @boolean
+     * @return
+     */
+    private Metadata getFullMetadataNoCache(final Contentlet contentlet,
+            final String fieldVariableName, final boolean forceGenerate) throws DotDataException {
         final StorageType storageType = StoragePersistenceProvider.getStorageType();
         final String metadataBucketName = Config.getStringProperty(METADATA_GROUP_NAME, DOT_METADATA);
         final String metadataPath = this.getFileName(contentlet, fieldVariableName);
 
-        return this.fileStorageAPI.retrieveMetaData(
-                new RequestMetadata.Builder()
+        Map<String, Serializable> metadataMap = fileStorageAPI.retrieveMetaData(
+                new FetchMetadataParams.Builder()
                         .cache(false)
-                        .wrapMetadataMapForCache(this::filterNonCacheableMetadataFields) // why is it needed for the non-cached version??
                         .storageKey(
-                                new StorageKey.Builder().group(metadataBucketName).path(metadataPath)
+                                new StorageKey.Builder().group(metadataBucketName)
+                                        .path(metadataPath)
                                         .storage(storageType).build())
                         .build()
         );
+
+        if(null != metadataMap){
+           return new Metadata(fieldVariableName, metadataMap);
+        }
+
+        if(forceGenerate){
```

Comment:
I think the previous comment applies here also

Reply:
I disagree